backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt

```diff
@@ -1026,6 +1026,7 @@ internal class CAdapterGenerator(val context: Context) : DeclarationDescriptorVi
             KonanPrimitiveType.FLOAT -> "${prefix}_KFloat"
             KonanPrimitiveType.DOUBLE -> "${prefix}_KDouble"
             KonanPrimitiveType.NON_NULL_NATIVE_PTR -> "void*"
+            KonanPrimitiveType.VECTOR16 -> "${prefix}_KVector"
```

Comment:
This is likely not enough, `*_KVector` type should also be defined.
Please add a test for this (see https://github.com/JetBrains/kotlin-native/tree/e1b3fefd3333fa730bb00174b7eb24c90822f6a0/backend.native/tests/produce_dynamic/simple).